integration-tests/tests/integration/main.rs

```diff
@@ -842,6 +843,113 @@ archive.include = [
         .expect_err("archive should have failed");
 }
 
+#[test]
+fn test_archive_with_build_filter() {
```

Comment:
This should also verify that we can run tests out of these stripped archives, and in particular that we don't try to run test binaries that aren't in the archive.